### DISCUSSION TITLE
Foxglove Data Platform Event Organization

### DISCUSSION THREAD

**Discussion by valschmidt** [foxglove/community#599] [Q&A]
As we are using Foxglove's Data platform more and more with different robots and many deployments we are finding the "Events" mechanism really useful for quickly locating logs of interest. But currently the space for viewing Events on the Data Platform is flat, which is making our events cluttered and confusing. We desperately need the ability to create folders and subfolders within which we can organize the Events, by vehicle, deployment or whatever. Tags are useful, but still make everything cluttered. Along with this is the ability to create and add/delete/move/rename filters, subfolders and the Events themselves from the CLI.

**Comment by amacneil** [comment]
Thanks for the feedback! We would rather not have _both_ tags and folders for events, but can you share more about why tags are not working for you today? 

E.g. events are already associated with a vehicle, is there a reason you would also want to create folders by vehicle?

Renaming/updating event tags is definitely a feature we plan to add.

Happy to screen share with you at some point to better understand this if it's easier.

**Reply by defunctzombie** [reply]
My read of the above is that "folders" are more like _views_ with filters of specific tags or search terms. This would make it easy to create and share views of specific searches like "incidents of X in last 30 days".

**Reply by valschmidt** [reply]
I will admit, folders are, in a sense, just another set of tags. 

But from a usability standpoint, I think they are fundamentally different, in that some librarian decides their name, the name is standardized and easy/obvious to enforce, and perhaps most importantly, they provide some visual organization to the events list page. An organization may have 10's of folks looking at data, and each will have their own tags specific to their interests. Invariably it will be a tag "wild-wild west", in which, for example, tags for deployment "NA142" are sometimes written na142, na-142, NA-142, NA_142 or "Deployment NA142", and searching for those events is a nightmare. Speaking from experience, some folks will be lazy, or just stubborn and their events won't get important categorical tags, any tags at all, or they will make up their own tags with every entry (which is worse than useless).

I realize some of this is social and organizational engineering. But I do think that the ability to provide some visual organization to the events would really help. 

There are a few other things that might help:

* When creating a new event tag, one should probably have the ability to either specify a new Key, or select one from a list previously specified, and then either specify a new Value, or again, select one from a list of those previously specified for that Key.  
* You should be able to add a text description to an event. 
* Another feature, beyond the scope of this conversation - would be a discussion board around an event. Maybe the easiest way to do this would be if Foxglove could provide a widget to provide hyperlinks to Events. Then one could host a conversation elsewhere (e.g. GitHub) and link the discussion to a particular event.

**Reply by amacneil** [reply]
Thanks! This is super helpful feedback.

We have been thinking about the "unstructured text" problem you describe. One solution I would like to explore is allowing the org admin to define a tag schema up front - e.g. certain fields could be "required", or could have input validation (e.g. `deployment` must be set to a value from a pre-approved list).
